src/Mod/TechDraw/Gui/QGTracker.cpp

```diff
@@ -438,12 +440,15 @@ void QGTracker::setPoint(std::vector<QPointF> pts)
         return;
     }
     prepareGeometryChange();
-    QPainterPath newPath;
-    QPointF center = pts.front();
-    double radius = 50.0;
-    newPath.addEllipse(center, radius, radius);
-    setPath(newPath);
-    setPrettyNormal();
+
+    auto point = new QGIVertex(-1);
+    point->setParentItem(this);
+    point->setPos(pts.front());
+    point->setRadius(dynamic_cast<QGIViewPart *>(m_qgParent)->getVertexSize());
```

Comment:
This should be `static_cast`.

In case we cannot expect the object to be `QGIViewPart`, we need a nullptr check else we would get a segmentation fault. Also, we should not use dynamic cast but Qt type check.

Reply:
Oh, right. This was a down cast so I used dynamic_cast but I get your reasoning that if I am expecting it to be something else then I need to handle the nullptr. Changing it to a static_casts means that we are sure that it won't be anything else right? Should I use a dynamic_cast and log an error if it is a nullptr?

Reply:
If it's a nullptr, then we cannot continue executing, as the rest of the function won't make sense: we need to abort/return, and send error message to console.

QGTracker adds lines and vertexes, and as far as I know, this can only be in QGIViewPart, so we should be sure it is always a QGIViewPart.

Reply:
I was thinking about using a approx value after logging in case it returns a nullptr.
But ig if it can't be anything else then that's pointless so I have updated the code to use static cast.